utils/parser.go

```diff
@@ -0,0 +1,46 @@
+package utils
+
+import (
+	"encoding/json"
+	"errors"
+	"log"
+)
+
+//reference human readable keys to DB key values
+var keyReferences = map[string]string{"id": "id", "createdDate": "created_at", "name": "name"}
+
+//ParseFilters is used to parse filters form JSON string to a map of (string,string)
+//If any required key is missing, it is set as default
+//All unknown keys are ignored
+func ParseFilters(filterString string) (map[string]string, error) {
+	if filterString == "" {
+		filterString = "{}"
+	}
+	filters := map[string]string{}
+	err := json.Unmarshal([]byte(filterString), &filters)
+	var cleanedFilters map[string]string
+	if err == nil {
+		cleanedFilters, err = cleanFilters(filters)
+	}
+	if err != nil {
+		log.Println(err)
+		return make(map[string]string), errors.New("Illegal filter syntax")
+	}
+	return cleanedFilters, nil
+}
+
+func cleanFilters(filters map[string]string) (map[string]string, error) {
+	supportedKeys := []string{"id", "createdDate", "name"}
```

Comment:
Magic strings should be in constants. When a string is being used in multiple places, pull them out into constants so that change in string value is isolated to a variable

Reply:
If you could suggest a proper place for it. That would be great